sympy/ntheory/tests/test_digits.py

```diff
@@ -1,8 +1,20 @@
-from sympy.ntheory import count_digits, digits, is_palindromic
+from sympy.ntheory import count_digits, digits, is_palindromic, ndigits
 
 from sympy.testing.pytest import raises
 
 
+def test_ndigits():
+    # depending on whether one rounds up or down or uses log or log10,
+    # one or more of these will fail if you don't check for the off-by
+    # one condition
+    assert ndigits(1000, 10) == 4
+    assert ndigits(125, 5) == 4
+    assert ndigits(100, 16) == 2
+    assert ndigits(-1000, 10) == 4
+    assert ndigits(-125, 5) == 4
+    assert ndigits(-100, 16) == 2
+
```

Comment:
```suggestion
    for base in range(2, 100):
        for e in range(1, 100):
            n = base**e
            assert ndigits(n, base) == e + 1
            assert ndigits(n + 1, base) == e + 1
            assert ndigits(n - 1, base) == e

```

Added tests

Reply:
I had run almost the exact same thing -- likely having been conditioned by reading your PRs ! But when I found these that broke the original algorithm for different rounding/log schemes I took them as a summary. I appreciate the feedback. This is a much more delicate problem than it appears! :-)